packages/react-loader/src/createQueryStore/universal.ts

```diff
@@ -40,6 +40,12 @@ export const createQueryStore = (
     params: QueryParams = {},
     options: Parameters<QueryStore['loadQuery']>[2] = {},
   ): Promise<QueryResponseInitial<QueryResponseResult>> => {
+    let stega: StegaConfig = {enabled: false}
+    if (options.stega) {
+      stega = options.stega === true ? {enabled: true} : options.stega
+    } else if (unstable__serverClient.instance?.config().stega) {
+      stega = unstable__serverClient.instance?.config().stega
+    }
```

Comment:
The `stega` option should only use the boolean short hand, as options given to `unstable__cache.instance.fetch` has to be strictly JSON serializable, and `stega.studioUrl`, `stega.filter`, and `stega.logger`, aren't.
```suggestion
    const stega = options.stega ?? unstable__serverClient.instance?.config().stega.enabled ?? false
```

Reply:
So then the type for `options` on 41 is incorrect? It suggests you can pass a `StegaConfig`

Reply:
Those `options` were typed for usage with `perspective: "previewDrafts"`, which doesn’t use `async-cache-dedupe` under the hood and thus isn’t limited to inputs that can be deserialized from strings and thus can use all the same stega options as `client.fetch`. 

I don’t know how to make TypeScript differentiate between the two but I’m open for suggestions

Reply:
A discriminated union?

Reply:
Has anything changed here? The latest version seem to include the suggested removal, is this safe to do now?

Reply:
I gave up trying to fix the typings (this is why I abandoned the PR the first time).

Oh and yeah I screwed up and merged the wrong suggestion 🤦 however Cody's suggestion above still potentially creates a `StegaConfig` type

```ts
const stega = options.stega ?? unstable__serverClient.instance?.config().stega.enabled ?? false
```

<img width="944" alt="Screenshot 2024-11-26 at 15 21 20" src="https://github.com/user-attachments/assets/0343afe1-0159-4e3f-befe-16adab137bf4">

I'm wondering now if it's supposed to be:

```ts
const stega = options.stega ? unstable__serverClient.instance?.config().stega.enabled : false
```

<img width="976" alt="Screenshot 2024-11-26 at 15 22 06" src="https://github.com/user-attachments/assets/7e637d34-5ca7-464c-aead-d5810e5c0bd5">

Reply:
Yep you're totally right about `StegaConfig` still being a possibility with the suggestion.

If we want to normalize to a boolean, can we just do this:

```suggestion
const stega =
  typeof options.stega === 'boolean'
    ? options.stega
    : (options.stega?.enabled ??
      unstable__serverClient.instance?.config().stega?.enabled ??
      false)
```

Reply:
It's filthy but I like it

Reply:
Please, just one more nested ternary nullish coalescence... I can stop whenever I want to.

@stipsan Is this correct behaviour to normalize to a bool? We good to go?

Reply:
Yes! Allowing other stega options on the request level opens up a world of pain 😮‍💨